### ISSUE TITLE
Error when Creating Custom Resource

### ISSUE THREAD

**Issue by Fortune117** [Facepunch/sbox-issues#2107]
### Describe the bug

When I try to make a custom resource with the following attribute: `[GameResource("Ammo", "ammo", "Data about a specific ammo item.")]`, I get this error:

![image](https://user-images.githubusercontent.com/7847372/182268181-27df1040-f193-42e6-b2b6-ce6d13a1fadb.png)
![image](https://user-images.githubusercontent.com/7847372/182268218-a6a5e81a-22db-4e2e-a275-d95319e985b6.png)

### To Reproduce

1. Add this resource to the game:
```CS
[GameResource("Ammo", "ammo", "Data about a specific ammo item.", Icon = "inventory")]
public class AmmoItemResource : GameResource
{
    [Category("Ammo Info")]
    public float Mass { get; set; }
    
    [Category("Ammo Info")]
    public float DragCoefficient { get; set; }
    
    [Category("Ammo Info")]
    public float CrossSectionalArea { get; set; } 
}
```
2. Create the resource in the asset browser.
3. Hopefully the error is reproduced. Otherwise, I'll see if I can make a repro.

### Expected behavior

The asset is created as expected.

### Additional context

When I changed the attribute to `[GameResource("Ammo Item two", "ammopoo", "Data about a specific ammo item.", Icon = "inventory")]` it seemed to work fine.

**Comment by Fortune117** [comment]
When I try and recompile an asset that references an `AmmoItemResource` it gives me this error now.:
![image](https://user-images.githubusercontent.com/7847372/182269751-40370382-32f8-40dc-a74f-82b751005480.png)

This is after I changed the resource extension and could create the asset.

**Comment by Fortune117** [comment]
After a little investigating, it seems most of my older resources are broken in the same way.
[resource code.zip](https://github.com/Facepunch/sbox-issues/files/9238895/resource.code.zip)

Here's a dump with everything you should need to load my resources. Hopefully it helps with reproduction.

**Comment by cr4yz** [comment]
Your first issue should be fixed with #2153 - could you create a new issue if your second issue still persists?

**Comment by Fortune117** [comment]
Yup this fixed it! Thanks.
